File: src/MediCore.API/MediCore.Domain/Entities/Inventory/InventoryItem.cs
using MediCore.Domain.Common;

namespace MediCore.Domain.Entities.Inventory;

public class InventoryItem : BaseEntity
{
    public Guid TenantId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Sku { get; set; } = string.Empty;
    public string? Barcode { get; set; }
    public Guid? CategoryId { get; set; }
    public string UnitOfMeasure { get; set; } = "unit";

    /// <summary>True for medications dispensed/sold through the Pharmacy module.</summary>
    public bool IsPharmacyItem { get; set; } = false;

    /// <summary>True when this item is tracked by batch/lot number and expiry date.</summary>
    public bool TracksBatches { get; set; } = false;

    public decimal CostPrice { get; set; }
    public decimal SalePrice { get; set; }
    public int ReorderPoint { get; set; } = 0;
    public int SafetyStock { get; set; } = 0;
    public bool IsActive { get; set; } = true;

    public InventoryCategory? Category { get; set; }
}

File: src/MediCore.Desktop/lib/features/inventory/domain/entities/inventory_item.dart
class InventoryItem {
  final String id;
  final String name;
  final String sku;
  final String? barcode;
  final String? categoryId;
  final String? categoryName;
  final String unitOfMeasure;
  final bool isPharmacyItem;
  final bool tracksBatches;
  final double costPrice;
  final double salePrice;
  final int reorderPoint;
  final int safetyStock;
  final bool isActive;

  const InventoryItem({
    required this.id,
    required this.name,
    required this.sku,
    this.barcode,
    this.categoryId,
    this.categoryName,
    required this.unitOfMeasure,
    required this.isPharmacyItem,
    required this.tracksBatches,
    required this.costPrice,
    required this.salePrice,
    required this.reorderPoint,
    required this.safetyStock,
    required this.isActive,
  });
}
